src/MLOps.NET.AWS/Storage/StorageAccountModelRepository.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Util;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.Storage
+{
+    internal sealed class StorageAccountModelRepository : IModelRepository
+    {
+        private readonly AmazonS3Client amazonS3Client;
+        private string bucketName;
+
+        public StorageAccountModelRepository(string awsAccessKeyId, string awsSecretAccessKey, string regionName, string bucketName)
+        {
+            var region = RegionEndpoint.GetBySystemName(regionName);
+            this.bucketName = bucketName;
+            this.amazonS3Client = new AmazonS3Client(awsAccessKeyId, awsSecretAccessKey, region);
+        }
+
+        public async Task UploadModelAsync(Guid runId, string filePath)
+        {
+            var bucketExists = await AmazonS3Util.DoesS3BucketExistV2Async(amazonS3Client, bucketName);
```

Comment:
Is there a non-static method we can use to check if a bucket exists or not? Having this static method hinders this method from being unit tested.